### ISSUE TITLE
(Feature Request) Add support for System SP, Sega Hikaru and Cave CV1000/CV1000D Hardware 

### ISSUE THREAD

**Issue by AbandonedWitchStarlight** [flyinghead/flycast#110]
To the developers of Flycast, I am requesting support for System SP, the Hikaru and the CAVE Cave CV1000/CV1000D Hardware, since they are variants of the Naomi Hardware. One emulator has those systems supported - but they aren't as complete and I am looking for proper Hikaru emulation.

**Comment by PiperCalls** [comment]
+1

**Comment by Shoegzer** [comment]
I posted a more detailed feature request about this [here](https://github.com/libretro/flycast/issues/418), on a separate (now defunct) repo, which has gotten some scrutiny from other devs.  I can repost it here if needed.

**Comment by dante3732** [comment]
Hikaru. 👍

**Comment by Iachino87** [comment]
plese hikaru!!!!!

**Comment by crashGG** [comment]
For Cave CV1000,  already have mame and fbneo, that can simulate 100% of the time.There is still Gaelco ,it is based on SH4, but more powerful than naomi1.The game is in mame’s gaelco/atvtrack.cpp driver list.

**Comment by flyinghead** [comment]
Dropping CAVE CV1000 in favor of Gaelco PowerVR-based games.

**Comment by Shoegzer** [comment]
> For Cave CV1000, already have mame and fbneo, that can simulate 100% of the time.

That's not correct. There are emulation inaccuracies that impact slowdown that Cave exploited to balance difficulty, as discussed [here](https://www.reddit.com/r/MAME/comments/prc6hc/if_anyone_asks_why_mame_isnt_recommended_for/) and other places.  It has gotten better, and the games are arguably in a "playable" state, but it's far from perfect.

**Comment by crashGG** [comment]
> > For Cave CV1000, already have mame and fbneo, that can simulate 100% of the time.
> 
> That's not correct. There are emulation inaccuracies that impact slowdown that Cave exploited to balance difficulty, as discussed [here](https://www.reddit.com/r/MAME/comments/prc6hc/if_anyone_asks_why_mame_isnt_recommended_for/) and other places. It has gotten better, and the games are arguably in a "playable" state, but it's far from perfect.
The discussion link you quoted is from 2 years ago and is out of date. If you follow such topics, you will find that impact slowdown has been accurately implemented in a commit six months.
https://github.com/mamedev/mame/pull/10849

**Comment by Shoegzer** [comment]
I understand the link is older, that's why I said it has gotten better since then.  That PR brings it closer to a playable state as I said, but as even its author indicates, it's far from perfect:

> This is still not perfect, and further work is needed on CV1000 slowdowns, especially on SH-3 CPU waitstates.

So again, your comment that CV1000 is "100%" is not correct.

**Comment by crashGG** [comment]
> I understand the link is older, that's why I said it has gotten better since then. That PR brings it closer to a playable state as I said, but as even its author indicates, it's far from perfect:
> 
> > This is still not perfect, and further work is needed on CV1000 slowdowns, especially on SH-3 CPU waitstates.
> 
> So again, your comment that CV1000 is "100%" is not correct.

The “100%” I wrote above is from the user’s perspective. If a game can run smoothly on a potato computer, with normal audio and video, and complete functions, users will exclaim: "This is 100% the fxcking game I remember!" From a developer's perspective, even if No matter how hard you try, can only get infinitely close, but will never reach 100%.

**Comment by Shoegzer** [comment]
Of course emulation can never achieve 100% accuracy, but that's not the point.  You're on GH interfacing with developers, not gamers.  Besides, even gamers that know CV1000 games well can easily identify the difference between MAME and the real thing, even with the new blitter improvements. Your 100% comment is misleading either way.

**Comment by crashGG** [comment]
IMHO，Based on the current level of Mame emulation, most games on cv1000 are hard to distinguish from real hardware.

**Comment by Shoegzer** [comment]
Again they are not hard to distinguish, but you believe what you want.

Regardless, stating something "can simulate 100% of the time" especially without qualifying what you mean as per your original post, is incorrect and misleading to users and developers alike.

This discussion will end now as it is off-topic from the intent of this and [related](https://github.com/libretro/flycast/issues/418) requests.

**Comment by barbudreadmon** [comment]
>Regardless, stating something "can simulate 100% of the time" especially without qualifying what you mean as per your original post, is incorrect and misleading to users and developers alike.

Regardless of the original intent of that "can simulate 100% of the time" (which seemed pretty clear to me), is Flycast likely to do a better job here ? More concretely, does it emulate cpu waitstates ?

**Comment by Thetouchedjoe** [comment]
bro we need Sega Hikaru and that Gaelco Power-VR games so i can have more games to play with the emulator

**Comment by arienti93** [comment]
@flyinghead good news about it?

**Comment by blueminder** [comment]
If there was news about it, this ticket would have been updated.
